lib/compressMeshes.js

```diff
@@ -0,0 +1,518 @@
+'use strict';
+var Cesium = require('cesium');
+var ForEach = require('./ForEach');
+var addExtensionsRequired = require('./addExtensionsRequired');
+var addToArray = require('./addToArray');
+var byteLengthForComponentType = require('./byteLengthForComponentType');
+var draco3d = require('draco3d');
+var getAccessorByteStride = require('./getAccessorByteStride');
+var getComponentReader = require('./getComponentReader.js');
+var hashObject = require('object-hash');
+var numberOfComponentsForType = require('./numberOfComponentsForType');
+
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+var arrayFill = Cesium.arrayFill;
+
+// Prepare encoder for compressing meshes.
+var encoderModule = draco3d.createEncoderModule({});
+
+module.exports = compressMeshes;
+
+function readAccessorPacked(gltf, accessor) {
+    var bufferView = gltf.bufferViews[accessor.bufferView];
+    var source = gltf.buffers[bufferView.buffer].extras._pipeline.source;
+    var numberOfComponents = numberOfComponentsForType(accessor.type);
+    var count = accessor.count;
+    var componentTypeByteLength = byteLengthForComponentType(accessor.componentType);
+    var byteOffset = accessor.byteOffset + bufferView.byteOffset + source.byteOffset;
+    var byteStride = getAccessorByteStride(gltf, accessor);
+
+    var dataView = new DataView(source.buffer);
+    var components = new Array(numberOfComponents);
+    var componentReader = getComponentReader(accessor.componentType);
+
+    var values = [];
+    for (var i = 0; i < count; ++i) {
+        componentReader(dataView, byteOffset, numberOfComponents, componentTypeByteLength, components);
+
+        for (var j = 0; j < numberOfComponents; ++j) {
+            values.push(components[j]);
+        }
+
+        byteOffset += byteStride;
+    }
+    return values;
+}
+
+/**
+ * Contains functions for removing elements from a glTF hierarchy.
+ * Since top-level glTF elements are arrays, when something is removed, referring
+ * indices need to be updated.
+ * @constructor
+ */
+function Remove() {}
+
+Remove.accessor = function(gltf, accessorId) {
```

Comment:
created file removedUnusedElements.js. For now this file exports one function that will remove "all" unused elements. This could always be refactored later if other code needed to call remove on one type at a time.